File: business-easy-core/src/main/java/com/businesseasy/core/entities/ProductEntity.java
package com.businesseasy.core.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "product", uniqueConstraints = {@UniqueConstraint(columnNames = {"name", "type", "brand", "country", "size"})})
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    Long id;

    @Column(name = "name")
    String name;

    @Column(name = "type")
    String type;

    @Column(name = "brand")
    String brand;

    @Column(name = "country")
    String country;

    @Column(name = "size")
    String size;
}

File: easiness-ui/app/server/entity/product.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, Unique } from "typeorm"
import { Base } from "./base.entity"


export const uniqueProductCols = ['name', 'type', 'brand', 'country', 'size']

@Entity({name: 'product'})
@Unique(uniqueProductCols)
export class Product extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column({nullable: true})
    type!: string

    @Column({nullable: true})
    brand!: string
    
    @Column()
    country: string

    @Column({nullable: true})
    size!: string

}